{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Bioconductor version '3.10' is out-of-date; the current release version '3.15'\n",
      "  is available with R version '4.2'; see https://bioconductor.org/install\n",
      "\n",
      "'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
      "'?repositories' for details\n",
      "\n",
      "replacement repositories:\n",
      "    CRAN: https://cran.r-project.org\n",
      "\n",
      "\n",
      "Bioconductor version 3.10 (BiocManager 1.30.19), R 3.6.1 (2019-07-05)\n",
      "\n",
      "Installing package(s) 'DESeq2'\n",
      "\n",
      "Warning message:\n",
      "“dependency ‘locfit’ is not available”\n",
      "also installing the dependencies ‘cli’, ‘lifecycle’, ‘fastmap’, ‘bit’, ‘vctrs’, ‘cachem’, ‘bitops’, ‘bit64’, ‘blob’, ‘memoise’, ‘Rcpp’, ‘deldir’, ‘RcppEigen’, ‘RCurl’, ‘zlibbioc’, ‘matrixStats’, ‘RSQLite’, ‘XML’, ‘png’, ‘jpeg’, ‘interp’, ‘checkmate’, ‘GenomeInfoDb’, ‘XVector’, ‘DelayedArray’, ‘AnnotationDbi’, ‘annotate’, ‘survival’, ‘latticeExtra’, ‘htmlTable’, ‘S4Vectors’, ‘IRanges’, ‘GenomicRanges’, ‘SummarizedExperiment’, ‘Biobase’, ‘BiocParallel’, ‘genefilter’, ‘geneplotter’, ‘Hmisc’, ‘RcppArmadillo’\n",
      "\n",
      "\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘cli’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘fastmap’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘bit’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘bitops’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘Rcpp’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘deldir’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘zlibbioc’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘matrixStats’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘XML’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘png’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘jpeg’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘checkmate’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘survival’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘S4Vectors’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘Biobase’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘BiocParallel’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘cachem’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘bit64’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘RcppEigen’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘RCurl’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘htmlTable’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘IRanges’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘RcppArmadillo’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘vctrs’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘memoise’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘interp’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘GenomeInfoDb’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘XVector’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘DelayedArray’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘blob’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘latticeExtra’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘GenomicRanges’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘RSQLite’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘SummarizedExperiment’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘Hmisc’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘AnnotationDbi’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘annotate’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘genefilter’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘geneplotter’ had non-zero exit status”\n",
      "Warning message in install.packages(...):\n",
      "“installation of package ‘DESeq2’ had non-zero exit status”\n",
      "Updating HTML index of packages in '.Library'\n",
      "\n",
      "Making 'packages.html' ...\n",
      " done\n",
      "\n",
      "Old packages: 'askpass', 'backports', 'BH', 'boot', 'broom', 'callr', 'caret',\n",
      "  'class', 'cli', 'clipr', 'cluster', 'codetools', 'colorspace', 'crayon',\n",
      "  'curl', 'data.table', 'dbplyr', 'dichromat', 'digest', 'dplyr', 'ellipsis',\n",
      "  'evaluate', 'fansi', 'forcats', 'foreach', 'formatR', 'fs', 'generics',\n",
      "  'ggplot2', 'glmnet', 'glue', 'gower', 'gtable', 'haven', 'hexbin', 'highr',\n",
      "  'hms', 'htmltools', 'htmlwidgets', 'httpuv', 'httr', 'ipred', 'IRdisplay',\n",
      "  'iterators', 'jsonlite', 'KernSmooth', 'knitr', 'labeling', 'later',\n",
      "  'lattice', 'lava', 'lubridate', 'magrittr', 'maps', 'markdown', 'MASS',\n",
      "  'Matrix', 'mgcv', 'mime', 'ModelMetrics', 'modelr', 'nlme', 'nnet',\n",
      "  'numDeriv', 'openssl', 'pbdZMQ', 'pillar', 'pkgconfig', 'plyr',\n",
      "  'prettyunits', 'processx', 'prodlim', 'progress', 'promises', 'ps', 'purrr',\n",
      "  'quantmod', 'R6', 'RColorBrewer', 'Rcpp', 'readr', 'readxl', 'recipes',\n",
      "  'repr', 'reprex', 'reshape2', 'rmarkdown', 'rpart', 'rstudioapi', 'rvest',\n",
      "  'scales', 'selectr', 'shiny', 'spatial', 'SQUAREM', 'stringi', 'stringr',\n",
      "  'survival', 'sys', 'tibble', 'tidyr', 'tidyselect', 'tidyverse', 'timeDate',\n",
      "  'tinytex', 'TTR', 'utf8', 'uuid', 'whisker', 'withr', 'xfun', 'xml2', 'xts',\n",
      "  'yaml', 'zoo'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if (!require(\"BiocManager\", quietly = TRUE))\n",
    "    install.packages(\"BiocManager\")\n",
    "\n",
    "BiocManager::install(\"DESeq2\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "Registered S3 methods overwritten by 'ggplot2':\n",
      "  method         from \n",
      "  [.quosures     rlang\n",
      "  c.quosures     rlang\n",
      "  print.quosures rlang\n",
      "\n",
      "Registered S3 method overwritten by 'rvest':\n",
      "  method            from\n",
      "  read_xml.response xml2\n",
      "\n",
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────── tidyverse 1.2.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.1.1     \u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 1.3.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 2.1.1     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.2\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 0.8.3     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.1.1     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.4.0\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in library(DESeq2): there is no package called ‘DESeq2’\n",
     "output_type": "error",
     "traceback": [
      "Error in library(DESeq2): there is no package called ‘DESeq2’\nTraceback:\n",
      "1. library(DESeq2)"
     ]
    }
   ],
   "source": [
    "library(dplyr)\n",
    "library(tidyverse)\n",
    "library(DESeq2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "countPATH <- \"/bigdata/faustmanlab/bch/testis_coculture_baseline/aligned/count_files\"\n",
    "\n",
    "files <- list.files(path=countPATH, pattern=\"*.tab\", full.names=TRUE)\n",
    "names(files) <- lapply(strsplit(files, \"/\"), \"[\",8)\n",
    "countslist <- lapply(files, read.delim, sep=\"\\t\", header=FALSE)\n",
    "names(countslist)<- names(files)\n",
    "colnames<- c(\"gene\", \"unstranded_counts\", \"Fstreadalign\",\"Sndreadalign\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>N_unmapped        </td><td>1338665           </td><td> 1338665          </td><td> 1338665          </td></tr>\n",
       "\t<tr><td>N_multimapping    </td><td>1754645           </td><td> 1754645          </td><td> 1754645          </td></tr>\n",
       "\t<tr><td>N_noFeature       </td><td>4424863           </td><td>17817452          </td><td>17609102          </td></tr>\n",
       "\t<tr><td>N_ambiguous       </td><td>1332592           </td><td>  186905          </td><td>  190919          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000066169</td><td>      0           </td><td>       0          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000070168</td><td>      0           </td><td>       0          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000070901</td><td>    560           </td><td>     276          </td><td>     284          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000018029</td><td>     43           </td><td>      28          </td><td>      15          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000031391</td><td>      5           </td><td>       5          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000055342</td><td>      0           </td><td>       0          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000068698</td><td>      3           </td><td>       3          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000067904</td><td>      0           </td><td>       0          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000063979</td><td>      0           </td><td>       0          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000064851</td><td>     37           </td><td>      21          </td><td>      17          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000063562</td><td>      0           </td><td>       0          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000050129</td><td>      0           </td><td>       0          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000056886</td><td>      0           </td><td>       0          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000032625</td><td>    287           </td><td>     132          </td><td>     155          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000029315</td><td>     30           </td><td>      20          </td><td>      10          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000016356</td><td>    690           </td><td>     320          </td><td>     370          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000028288</td><td>      0           </td><td>       0          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000015810</td><td>   1204           </td><td>     596          </td><td>     618          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000017538</td><td>    149           </td><td>      75          </td><td>      74          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000027204</td><td>    171           </td><td>      83          </td><td>      88          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000058651</td><td>      0           </td><td>       0          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000070886</td><td>     20           </td><td>      23          </td><td>      18          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000056005</td><td>      0           </td><td>       0          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000035453</td><td>      0           </td><td>       0          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000063396</td><td>      0           </td><td>       0          </td><td>       0          </td></tr>\n",
       "\t<tr><td>ENSRNOG00000068709</td><td>      0           </td><td>     250          </td><td>     265          </td></tr>\n",
       "\t<tr><td>...</td><td>...</td><td>...</td><td>...</td></tr>\n",
       "\t<tr><td>ENSRNOG00000066592</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000065193</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000069132</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000070299</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000068736</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000065175</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000062672</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000062596</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000066684</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000067500</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000067927</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000062558</td><td> 49               </td><td> 31               </td><td> 18               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000069483</td><td> 75               </td><td> 30               </td><td> 45               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000062942</td><td> 75               </td><td> 40               </td><td> 35               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000070031</td><td>  5               </td><td>428               </td><td>418               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000069827</td><td>195               </td><td> 80               </td><td>115               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000069226</td><td>607               </td><td>740               </td><td>708               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000068289</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000062439</td><td> 47               </td><td> 31               </td><td> 22               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000066356</td><td>185               </td><td> 90               </td><td>101               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000068919</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000068656</td><td>  5               </td><td>  2               </td><td>  3               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000064259</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000065062</td><td>142               </td><td> 88               </td><td> 54               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000062442</td><td>127               </td><td> 86               </td><td> 69               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000070895</td><td> 53               </td><td> 34               </td><td> 47               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000066963</td><td> 49               </td><td> 21               </td><td> 28               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000065223</td><td>  0               </td><td>  0               </td><td>  0               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000069230</td><td>  9               </td><td> 84               </td><td>107               </td></tr>\n",
       "\t<tr><td>ENSRNOG00000071103</td><td> 31               </td><td> 77               </td><td> 75               </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llll}\n",
       " V1 & V2 & V3 & V4\\\\\n",
       "\\hline\n",
       "\t N\\_unmapped         & 1338665              &  1338665             &  1338665            \\\\\n",
       "\t N\\_multimapping     & 1754645              &  1754645             &  1754645            \\\\\n",
       "\t N\\_noFeature        & 4424863              & 17817452             & 17609102            \\\\\n",
       "\t N\\_ambiguous        & 1332592              &   186905             &   190919            \\\\\n",
       "\t ENSRNOG00000066169 &       0            &        0           &        0          \\\\\n",
       "\t ENSRNOG00000070168 &       0            &        0           &        0          \\\\\n",
       "\t ENSRNOG00000070901 &     560            &      276           &      284          \\\\\n",
       "\t ENSRNOG00000018029 &      43            &       28           &       15          \\\\\n",
       "\t ENSRNOG00000031391 &       5            &        5           &        0          \\\\\n",
       "\t ENSRNOG00000055342 &       0            &        0           &        0          \\\\\n",
       "\t ENSRNOG00000068698 &       3            &        3           &        0          \\\\\n",
       "\t ENSRNOG00000067904 &       0            &        0           &        0          \\\\\n",
       "\t ENSRNOG00000063979 &       0            &        0           &        0          \\\\\n",
       "\t ENSRNOG00000064851 &      37            &       21           &       17          \\\\\n",
       "\t ENSRNOG00000063562 &       0            &        0           &        0          \\\\\n",
       "\t ENSRNOG00000050129 &       0            &        0           &        0          \\\\\n",
       "\t ENSRNOG00000056886 &       0            &        0           &        0          \\\\\n",
       "\t ENSRNOG00000032625 &     287            &      132           &      155          \\\\\n",
       "\t ENSRNOG00000029315 &      30            &       20           &       10          \\\\\n",
       "\t ENSRNOG00000016356 &     690            &      320           &      370          \\\\\n",
       "\t ENSRNOG00000028288 &       0            &        0           &        0          \\\\\n",
       "\t ENSRNOG00000015810 &    1204            &      596           &      618          \\\\\n",
       "\t ENSRNOG00000017538 &     149            &       75           &       74          \\\\\n",
       "\t ENSRNOG00000027204 &     171            &       83           &       88          \\\\\n",
       "\t ENSRNOG00000058651 &       0            &        0           &        0          \\\\\n",
       "\t ENSRNOG00000070886 &      20            &       23           &       18          \\\\\n",
       "\t ENSRNOG00000056005 &       0            &        0           &        0          \\\\\n",
       "\t ENSRNOG00000035453 &       0            &        0           &        0          \\\\\n",
       "\t ENSRNOG00000063396 &       0            &        0           &        0          \\\\\n",
       "\t ENSRNOG00000068709 &       0            &      250           &      265          \\\\\n",
       "\t ... & ... & ... & ...\\\\\n",
       "\t ENSRNOG00000066592 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000065193 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000069132 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000070299 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000068736 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000065175 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000062672 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000062596 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000066684 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000067500 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000067927 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000062558 &  49                &  31                &  18               \\\\\n",
       "\t ENSRNOG00000069483 &  75                &  30                &  45               \\\\\n",
       "\t ENSRNOG00000062942 &  75                &  40                &  35               \\\\\n",
       "\t ENSRNOG00000070031 &   5                & 428                & 418               \\\\\n",
       "\t ENSRNOG00000069827 & 195                &  80                & 115               \\\\\n",
       "\t ENSRNOG00000069226 & 607                & 740                & 708               \\\\\n",
       "\t ENSRNOG00000068289 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000062439 &  47                &  31                &  22               \\\\\n",
       "\t ENSRNOG00000066356 & 185                &  90                & 101               \\\\\n",
       "\t ENSRNOG00000068919 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000068656 &   5                &   2                &   3               \\\\\n",
       "\t ENSRNOG00000064259 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000065062 & 142                &  88                &  54               \\\\\n",
       "\t ENSRNOG00000062442 & 127                &  86                &  69               \\\\\n",
       "\t ENSRNOG00000070895 &  53                &  34                &  47               \\\\\n",
       "\t ENSRNOG00000066963 &  49                &  21                &  28               \\\\\n",
       "\t ENSRNOG00000065223 &   0                &   0                &   0               \\\\\n",
       "\t ENSRNOG00000069230 &   9                &  84                & 107               \\\\\n",
       "\t ENSRNOG00000071103 &  31                &  77                &  75               \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| V1 | V2 | V3 | V4 |\n",
       "|---|---|---|---|\n",
       "| N_unmapped         | 1338665            |  1338665           |  1338665           |\n",
       "| N_multimapping     | 1754645            |  1754645           |  1754645           |\n",
       "| N_noFeature        | 4424863            | 17817452           | 17609102           |\n",
       "| N_ambiguous        | 1332592            |   186905           |   190919           |\n",
       "| ENSRNOG00000066169 |       0            |        0           |        0           |\n",
       "| ENSRNOG00000070168 |       0            |        0           |        0           |\n",
       "| ENSRNOG00000070901 |     560            |      276           |      284           |\n",
       "| ENSRNOG00000018029 |      43            |       28           |       15           |\n",
       "| ENSRNOG00000031391 |       5            |        5           |        0           |\n",
       "| ENSRNOG00000055342 |       0            |        0           |        0           |\n",
       "| ENSRNOG00000068698 |       3            |        3           |        0           |\n",
       "| ENSRNOG00000067904 |       0            |        0           |        0           |\n",
       "| ENSRNOG00000063979 |       0            |        0           |        0           |\n",
       "| ENSRNOG00000064851 |      37            |       21           |       17           |\n",
       "| ENSRNOG00000063562 |       0            |        0           |        0           |\n",
       "| ENSRNOG00000050129 |       0            |        0           |        0           |\n",
       "| ENSRNOG00000056886 |       0            |        0           |        0           |\n",
       "| ENSRNOG00000032625 |     287            |      132           |      155           |\n",
       "| ENSRNOG00000029315 |      30            |       20           |       10           |\n",
       "| ENSRNOG00000016356 |     690            |      320           |      370           |\n",
       "| ENSRNOG00000028288 |       0            |        0           |        0           |\n",
       "| ENSRNOG00000015810 |    1204            |      596           |      618           |\n",
       "| ENSRNOG00000017538 |     149            |       75           |       74           |\n",
       "| ENSRNOG00000027204 |     171            |       83           |       88           |\n",
       "| ENSRNOG00000058651 |       0            |        0           |        0           |\n",
       "| ENSRNOG00000070886 |      20            |       23           |       18           |\n",
       "| ENSRNOG00000056005 |       0            |        0           |        0           |\n",
       "| ENSRNOG00000035453 |       0            |        0           |        0           |\n",
       "| ENSRNOG00000063396 |       0            |        0           |        0           |\n",
       "| ENSRNOG00000068709 |       0            |      250           |      265           |\n",
       "| ... | ... | ... | ... |\n",
       "| ENSRNOG00000066592 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000065193 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000069132 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000070299 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000068736 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000065175 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000062672 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000062596 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000066684 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000067500 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000067927 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000062558 |  49                |  31                |  18                |\n",
       "| ENSRNOG00000069483 |  75                |  30                |  45                |\n",
       "| ENSRNOG00000062942 |  75                |  40                |  35                |\n",
       "| ENSRNOG00000070031 |   5                | 428                | 418                |\n",
       "| ENSRNOG00000069827 | 195                |  80                | 115                |\n",
       "| ENSRNOG00000069226 | 607                | 740                | 708                |\n",
       "| ENSRNOG00000068289 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000062439 |  47                |  31                |  22                |\n",
       "| ENSRNOG00000066356 | 185                |  90                | 101                |\n",
       "| ENSRNOG00000068919 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000068656 |   5                |   2                |   3                |\n",
       "| ENSRNOG00000064259 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000065062 | 142                |  88                |  54                |\n",
       "| ENSRNOG00000062442 | 127                |  86                |  69                |\n",
       "| ENSRNOG00000070895 |  53                |  34                |  47                |\n",
       "| ENSRNOG00000066963 |  49                |  21                |  28                |\n",
       "| ENSRNOG00000065223 |   0                |   0                |   0                |\n",
       "| ENSRNOG00000069230 |   9                |  84                | 107                |\n",
       "| ENSRNOG00000071103 |  31                |  77                |  75                |\n",
       "\n"
      ],
      "text/plain": [
       "      V1                 V2      V3       V4      \n",
       "1     N_unmapped         1338665  1338665  1338665\n",
       "2     N_multimapping     1754645  1754645  1754645\n",
       "3     N_noFeature        4424863 17817452 17609102\n",
       "4     N_ambiguous        1332592   186905   190919\n",
       "5     ENSRNOG00000066169       0        0        0\n",
       "6     ENSRNOG00000070168       0        0        0\n",
       "7     ENSRNOG00000070901     560      276      284\n",
       "8     ENSRNOG00000018029      43       28       15\n",
       "9     ENSRNOG00000031391       5        5        0\n",
       "10    ENSRNOG00000055342       0        0        0\n",
       "11    ENSRNOG00000068698       3        3        0\n",
       "12    ENSRNOG00000067904       0        0        0\n",
       "13    ENSRNOG00000063979       0        0        0\n",
       "14    ENSRNOG00000064851      37       21       17\n",
       "15    ENSRNOG00000063562       0        0        0\n",
       "16    ENSRNOG00000050129       0        0        0\n",
       "17    ENSRNOG00000056886       0        0        0\n",
       "18    ENSRNOG00000032625     287      132      155\n",
       "19    ENSRNOG00000029315      30       20       10\n",
       "20    ENSRNOG00000016356     690      320      370\n",
       "21    ENSRNOG00000028288       0        0        0\n",
       "22    ENSRNOG00000015810    1204      596      618\n",
       "23    ENSRNOG00000017538     149       75       74\n",
       "24    ENSRNOG00000027204     171       83       88\n",
       "25    ENSRNOG00000058651       0        0        0\n",
       "26    ENSRNOG00000070886      20       23       18\n",
       "27    ENSRNOG00000056005       0        0        0\n",
       "28    ENSRNOG00000035453       0        0        0\n",
       "29    ENSRNOG00000063396       0        0        0\n",
       "30    ENSRNOG00000068709       0      250      265\n",
       "...   ...                ...     ...      ...     \n",
       "30535 ENSRNOG00000066592   0       0        0     \n",
       "30536 ENSRNOG00000065193   0       0        0     \n",
       "30537 ENSRNOG00000069132   0       0        0     \n",
       "30538 ENSRNOG00000070299   0       0        0     \n",
       "30539 ENSRNOG00000068736   0       0        0     \n",
       "30540 ENSRNOG00000065175   0       0        0     \n",
       "30541 ENSRNOG00000062672   0       0        0     \n",
       "30542 ENSRNOG00000062596   0       0        0     \n",
       "30543 ENSRNOG00000066684   0       0        0     \n",
       "30544 ENSRNOG00000067500   0       0        0     \n",
       "30545 ENSRNOG00000067927   0       0        0     \n",
       "30546 ENSRNOG00000062558  49      31       18     \n",
       "30547 ENSRNOG00000069483  75      30       45     \n",
       "30548 ENSRNOG00000062942  75      40       35     \n",
       "30549 ENSRNOG00000070031   5     428      418     \n",
       "30550 ENSRNOG00000069827 195      80      115     \n",
       "30551 ENSRNOG00000069226 607     740      708     \n",
       "30552 ENSRNOG00000068289   0       0        0     \n",
       "30553 ENSRNOG00000062439  47      31       22     \n",
       "30554 ENSRNOG00000066356 185      90      101     \n",
       "30555 ENSRNOG00000068919   0       0        0     \n",
       "30556 ENSRNOG00000068656   5       2        3     \n",
       "30557 ENSRNOG00000064259   0       0        0     \n",
       "30558 ENSRNOG00000065062 142      88       54     \n",
       "30559 ENSRNOG00000062442 127      86       69     \n",
       "30560 ENSRNOG00000070895  53      34       47     \n",
       "30561 ENSRNOG00000066963  49      21       28     \n",
       "30562 ENSRNOG00000065223   0       0        0     \n",
       "30563 ENSRNOG00000069230   9      84      107     \n",
       "30564 ENSRNOG00000071103  31      77       75     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "countslist[[1]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>A_827_S1_ReadsPerGene.out.tab</th><th scope=col>B_827_S2_ReadsPerGene.out.tab</th><th scope=col>C_827_S3_ReadsPerGene.out.tab</th><th scope=col>D1_A2_Day3_ReadsPerGene.out.tab</th><th scope=col>D1_A3_Day5_ReadsPerGene.out.tab</th><th scope=col>D1_A4_Day7_ReadsPerGene.out.tab</th><th scope=col>D2_A3_Day5_ReadsPerGene.out.tab</th><th scope=col>D2_A4_Day7_ReadsPerGene.out.tab</th><th scope=col>D3_A2_Day3_ReadsPerGene.out.tab</th><th scope=col>D3_A3_Day5_ReadsPerGene.out.tab</th><th scope=col>D3_A4_Day7_ReadsPerGene.out.tab</th><th scope=col>D4_A2_Day3_ReadsPerGene.out.tab</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>ENSRNOG00000066169</th><td>  0 </td><td>  7 </td><td>  6 </td><td>   4</td><td>  0 </td><td>  3 </td><td>   0</td><td>   6</td><td>   5</td><td>  0 </td><td>   2</td><td>   4</td></tr>\n",
       "\t<tr><th scope=row>ENSRNOG00000070168</th><td>  0 </td><td>  0 </td><td>  0 </td><td>   0</td><td>  0 </td><td>  0 </td><td>   0</td><td>   0</td><td>   0</td><td>  0 </td><td>   0</td><td>   0</td></tr>\n",
       "\t<tr><th scope=row>ENSRNOG00000070901</th><td>560 </td><td>709 </td><td>747 </td><td>1383</td><td>928 </td><td>913 </td><td>1489</td><td>1273</td><td>1691</td><td>909 </td><td>1020</td><td>1086</td></tr>\n",
       "\t<tr><th scope=row>ENSRNOG00000018029</th><td> 43 </td><td> 35 </td><td> 32 </td><td>  37</td><td> 20 </td><td> 24 </td><td>  34</td><td>  48</td><td>  33</td><td> 18 </td><td>  25</td><td>  51</td></tr>\n",
       "\t<tr><th scope=row>ENSRNOG00000031391</th><td>  5 </td><td> 29 </td><td> 35 </td><td>   0</td><td>  0 </td><td>  1 </td><td>   2</td><td>   0</td><td>   2</td><td>  0 </td><td>   0</td><td>   0</td></tr>\n",
       "\t<tr><th scope=row>ENSRNOG00000055342</th><td>  0 </td><td>  0 </td><td>  0 </td><td>   0</td><td>  0 </td><td>  0 </td><td>   0</td><td>   0</td><td>   0</td><td>  0 </td><td>   0</td><td>   0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & A\\_827\\_S1\\_ReadsPerGene.out.tab & B\\_827\\_S2\\_ReadsPerGene.out.tab & C\\_827\\_S3\\_ReadsPerGene.out.tab & D1\\_A2\\_Day3\\_ReadsPerGene.out.tab & D1\\_A3\\_Day5\\_ReadsPerGene.out.tab & D1\\_A4\\_Day7\\_ReadsPerGene.out.tab & D2\\_A3\\_Day5\\_ReadsPerGene.out.tab & D2\\_A4\\_Day7\\_ReadsPerGene.out.tab & D3\\_A2\\_Day3\\_ReadsPerGene.out.tab & D3\\_A3\\_Day5\\_ReadsPerGene.out.tab & D3\\_A4\\_Day7\\_ReadsPerGene.out.tab & D4\\_A2\\_Day3\\_ReadsPerGene.out.tab\\\\\n",
       "\\hline\n",
       "\tENSRNOG00000066169 &   0  &   7  &   6  &    4 &   0  &   3  &    0 &    6 &    5 &   0  &    2 &    4\\\\\n",
       "\tENSRNOG00000070168 &   0  &   0  &   0  &    0 &   0  &   0  &    0 &    0 &    0 &   0  &    0 &    0\\\\\n",
       "\tENSRNOG00000070901 & 560  & 709  & 747  & 1383 & 928  & 913  & 1489 & 1273 & 1691 & 909  & 1020 & 1086\\\\\n",
       "\tENSRNOG00000018029 &  43  &  35  &  32  &   37 &  20  &  24  &   34 &   48 &   33 &  18  &   25 &   51\\\\\n",
       "\tENSRNOG00000031391 &   5  &  29  &  35  &    0 &   0  &   1  &    2 &    0 &    2 &   0  &    0 &    0\\\\\n",
       "\tENSRNOG00000055342 &   0  &   0  &   0  &    0 &   0  &   0  &    0 &    0 &    0 &   0  &    0 &    0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | A_827_S1_ReadsPerGene.out.tab | B_827_S2_ReadsPerGene.out.tab | C_827_S3_ReadsPerGene.out.tab | D1_A2_Day3_ReadsPerGene.out.tab | D1_A3_Day5_ReadsPerGene.out.tab | D1_A4_Day7_ReadsPerGene.out.tab | D2_A3_Day5_ReadsPerGene.out.tab | D2_A4_Day7_ReadsPerGene.out.tab | D3_A2_Day3_ReadsPerGene.out.tab | D3_A3_Day5_ReadsPerGene.out.tab | D3_A4_Day7_ReadsPerGene.out.tab | D4_A2_Day3_ReadsPerGene.out.tab |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSRNOG00000066169 |   0  |   7  |   6  |    4 |   0  |   3  |    0 |    6 |    5 |   0  |    2 |    4 |\n",
       "| ENSRNOG00000070168 |   0  |   0  |   0  |    0 |   0  |   0  |    0 |    0 |    0 |   0  |    0 |    0 |\n",
       "| ENSRNOG00000070901 | 560  | 709  | 747  | 1383 | 928  | 913  | 1489 | 1273 | 1691 | 909  | 1020 | 1086 |\n",
       "| ENSRNOG00000018029 |  43  |  35  |  32  |   37 |  20  |  24  |   34 |   48 |   33 |  18  |   25 |   51 |\n",
       "| ENSRNOG00000031391 |   5  |  29  |  35  |    0 |   0  |   1  |    2 |    0 |    2 |   0  |    0 |    0 |\n",
       "| ENSRNOG00000055342 |   0  |   0  |   0  |    0 |   0  |   0  |    0 |    0 |    0 |   0  |    0 |    0 |\n",
       "\n"
      ],
      "text/plain": [
       "                   A_827_S1_ReadsPerGene.out.tab B_827_S2_ReadsPerGene.out.tab\n",
       "ENSRNOG00000066169   0                             7                          \n",
       "ENSRNOG00000070168   0                             0                          \n",
       "ENSRNOG00000070901 560                           709                          \n",
       "ENSRNOG00000018029  43                            35                          \n",
       "ENSRNOG00000031391   5                            29                          \n",
       "ENSRNOG00000055342   0                             0                          \n",
       "                   C_827_S3_ReadsPerGene.out.tab\n",
       "ENSRNOG00000066169   6                          \n",
       "ENSRNOG00000070168   0                          \n",
       "ENSRNOG00000070901 747                          \n",
       "ENSRNOG00000018029  32                          \n",
       "ENSRNOG00000031391  35                          \n",
       "ENSRNOG00000055342   0                          \n",
       "                   D1_A2_Day3_ReadsPerGene.out.tab\n",
       "ENSRNOG00000066169    4                           \n",
       "ENSRNOG00000070168    0                           \n",
       "ENSRNOG00000070901 1383                           \n",
       "ENSRNOG00000018029   37                           \n",
       "ENSRNOG00000031391    0                           \n",
       "ENSRNOG00000055342    0                           \n",
       "                   D1_A3_Day5_ReadsPerGene.out.tab\n",
       "ENSRNOG00000066169   0                            \n",
       "ENSRNOG00000070168   0                            \n",
       "ENSRNOG00000070901 928                            \n",
       "ENSRNOG00000018029  20                            \n",
       "ENSRNOG00000031391   0                            \n",
       "ENSRNOG00000055342   0                            \n",
       "                   D1_A4_Day7_ReadsPerGene.out.tab\n",
       "ENSRNOG00000066169   3                            \n",
       "ENSRNOG00000070168   0                            \n",
       "ENSRNOG00000070901 913                            \n",
       "ENSRNOG00000018029  24                            \n",
       "ENSRNOG00000031391   1                            \n",
       "ENSRNOG00000055342   0                            \n",
       "                   D2_A3_Day5_ReadsPerGene.out.tab\n",
       "ENSRNOG00000066169    0                           \n",
       "ENSRNOG00000070168    0                           \n",
       "ENSRNOG00000070901 1489                           \n",
       "ENSRNOG00000018029   34                           \n",
       "ENSRNOG00000031391    2                           \n",
       "ENSRNOG00000055342    0                           \n",
       "                   D2_A4_Day7_ReadsPerGene.out.tab\n",
       "ENSRNOG00000066169    6                           \n",
       "ENSRNOG00000070168    0                           \n",
       "ENSRNOG00000070901 1273                           \n",
       "ENSRNOG00000018029   48                           \n",
       "ENSRNOG00000031391    0                           \n",
       "ENSRNOG00000055342    0                           \n",
       "                   D3_A2_Day3_ReadsPerGene.out.tab\n",
       "ENSRNOG00000066169    5                           \n",
       "ENSRNOG00000070168    0                           \n",
       "ENSRNOG00000070901 1691                           \n",
       "ENSRNOG00000018029   33                           \n",
       "ENSRNOG00000031391    2                           \n",
       "ENSRNOG00000055342    0                           \n",
       "                   D3_A3_Day5_ReadsPerGene.out.tab\n",
       "ENSRNOG00000066169   0                            \n",
       "ENSRNOG00000070168   0                            \n",
       "ENSRNOG00000070901 909                            \n",
       "ENSRNOG00000018029  18                            \n",
       "ENSRNOG00000031391   0                            \n",
       "ENSRNOG00000055342   0                            \n",
       "                   D3_A4_Day7_ReadsPerGene.out.tab\n",
       "ENSRNOG00000066169    2                           \n",
       "ENSRNOG00000070168    0                           \n",
       "ENSRNOG00000070901 1020                           \n",
       "ENSRNOG00000018029   25                           \n",
       "ENSRNOG00000031391    0                           \n",
       "ENSRNOG00000055342    0                           \n",
       "                   D4_A2_Day3_ReadsPerGene.out.tab\n",
       "ENSRNOG00000066169    4                           \n",
       "ENSRNOG00000070168    0                           \n",
       "ENSRNOG00000070901 1086                           \n",
       "ENSRNOG00000018029   51                           \n",
       "ENSRNOG00000031391    0                           \n",
       "ENSRNOG00000055342    0                           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# check that this is proper column\n",
    "\n",
    "for (i in seq_along(countslist)){\n",
    "  colnames(countslist[[i]]) <- colnames}\n",
    "\n",
    "counts_temp <- lapply(countslist, function(x) {x <- x %>% dplyr::select(-c(Fstreadalign,Sndreadalign))})\n",
    "counts_use <- lapply(counts_temp, function(x) {x <- x[-c(1:4),]})\n",
    "all_counts <- reduce(counts_use, full_join, by = \"gene\")\n",
    "colnames(all_counts)<-c(\"gene\", names(counts_use))\n",
    "\n",
    "rownames(all_counts)<- all_counts$gene\n",
    "all_counts <- all_counts[,-1]\n",
    "\n",
    "\n",
    "\n",
    "head(all_counts)\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "6e3c4fc0950aa63ef7bc9a4700bf9637652f40f7a53372464e6456fc41142b34"
  },
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
