---
title: "RNAseq_counts"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown


REVIEW THIS: http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#differential-expression-analysis


This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

## Including Plots

You can also embed plots, for example:

```{r pressure, include=FALSE}
library(dplyr)
library(tidyverse)
library(data.table)
library(DESeq2)
library(Glimma)
library(biomaRt)

```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.



```{r load counts}


countPATH <- "/bigdata/faustmanlab/bch/testis_coculture_baseline/aligned/count_files"

files <- list.files(path=countPATH, pattern="*.tab", full.names=TRUE)
names(files) <- lapply(strsplit(files, "/"), "[",8)
countslist <- lapply(files, read.delim, sep="\t", header=FALSE)
names(countslist)<- names(files)
colnames<- c("gene", "unstranded_counts", "Fstreadalign","Sndreadalign")


```


```{r rename}

# check that this is proper column


for (i in seq_along(countslist)){
  colnames(countslist[[i]]) <- colnames}

counts_temp <- lapply(countslist, function(x) {x <- x %>% dplyr::select(-c(Fstreadalign,Sndreadalign))})
counts_use <- lapply(counts_temp, function(x) {x <- x[-c(1:4),]})
all_counts <- plyr::join_all(counts_use, by = "gene")
colnames(all_counts)<-c("gene", names(counts_use))

rownames(all_counts)<- all_counts$gene
all_counts <- all_counts[,-1]



head(all_counts)

```
```{r metadata}

metadata <- data.table(

  sampleID=colnames(all_counts),
  samplename=paste0(sapply(strsplit(colnames(all_counts),"_"), `[`, 1),sapply(strsplit(colnames(all_counts),"_"), `[`, 2), sapply(strsplit(colnames(all_counts),"_"), `[`, 3)),
  date=sapply(strsplit(colnames(all_counts),"_"), `[`, 3),
  dish=sapply(strsplit(colnames(all_counts),"_"), `[`, 1),
  well=sapply(strsplit(colnames(all_counts),"_"), `[`, 2)
  
)

metadata$date[c(1:3)] <- "prep"

metadata$date <- as.factor(metadata$date)

relevel(metadata$date, ref = "prep")

```

```{r deseq2}

dds <- DESeqDataSetFromMatrix(countData=all_counts,
                                colData = metadata,
                                 design = ~date)





dds <- DESeq(dds)
```
```{r min-counts}

keep <- rowSums(counts(dds) >= 5) >= 3
table(keep)



```


```{r filt genes}

dds <- dds[keep,]


```


```{r boxplot}

dds <- estimateSizeFactors(dds)
boxplot(log10(counts(dds,normalized=TRUE)+1))
```



```{r deseq2 workup}

res <- results(dds)
head(results(dds))
summary(res)

```

```{r pca}

vsdata <- vst(dds, blind=FALSE)
plotPCA(vsdata, intgroup="date")

```
```{r glimma}



#glimmaMDS(dds)


```



```{r glimma ma}


head(res[order(res$pvalue),])
```


```{r plotMA}

plotMA(res, ylim=c(-5,5))

resultsNames(dds)
```




```{r function results table}


contrast_function <- function(object, parameter, contrast1, contrast2){
  
  x <- as.data.frame(results( object, contrast=c(parameter,contrast1,contrast2)))

x$geneID <- rownames(x)

mart <- useDataset("rnorvegicus_gene_ensembl", useMart("ensembl"))
genes <- rownames(x)

glist <- getBM(filters= "ensembl_gene_id", attributes=c("ensembl_gene_id","rgd_symbol"),values=genes,mart= mart)
y <- merge(x,glist,by.x="geneID",by.y="ensembl_gene_id")

return(y) }

```

```{r day7 to prep}

day7_to_prep <- contrast_function(dds,"date","Day7","prep")

```

```{r day7 to day3}

day7_to_day3 <- contrast_function(dds,"date","Day7","Day3")

```

```{r day7 to day5}

day7_to_day5 <- contrast_function(dds,"date","Day7","Day5")

```

```{r plot genes}

a <- plotCounts(dds, gene=("ENSRNOG00000015075"), intgroup="date", returnData=TRUE)
library("ggplot2")
ggplot(a, aes(x=date, y=count)) + 
  geom_point(position=position_jitter(w=0.1,h=0)) 
  #scale_y_log10(breaks=c(25,100,400))

## https://academic.oup.com/biolreprod/article/82/1/54/2557859#60652804

```

